### PR TITLE
STAC-8617: Include new tracer and centralize logic to build namespace…

### PR DESCRIPTION
**Pull request by craffit**
- Use new tracer
- Move logic for namespace generation based on loopback to this process agent
- Drop EndpointId etc from protobuf, it is not used in the protocol
- 'Deprecate' the scope field (it does not exist anymore in the agent, in sts it will be deprecated in another mr), namespaces are what is used now to determine where a connection lives

### CODE REVIEW

File: checks/net_common.go [hunk 1]
```diff
@@ -61,44 +74,60 @@ func createRelationIdentifier(localEndpoint, remoteEndpoint *model.EndpointId, d
...
-	if n != "" {
-		fragments = append(fragments, n)
+
+	// In order to tell different pod-local ip addresses from each other,
+	// treat each loopback address as local to the network namespace
+	// Reference implementation: https://github.com/weaveworks/scope/blob/master/report/id.go#L40
+	// https://github.com/weaveworks/scope/blob/7163f42170d72702fd55d2324d203c5b7be5c5cc/probe/endpoint/ebpf.go#L34
+	// We disregard local ip addresses for now, those might be interesting when doing docker setups,
+	// which are not the highest priority atm
+	if (isLoopback(connection.Local) || isLoopback(connection.Remote)) {
```
Comment (craffit) [inline checks/net_common.go:112]: make this conjunction

File: Gopkg.lock [hunk 2]
```diff
@@ -1142,6 +1141,7 @@
     "github.com/gogo/protobuf/jsonpb",
     "github.com/gogo/protobuf/proto",
     "github.com/mailru/easyjson",
+    "github.com/patrickmn/go-cache",
```
Comment (TizianoPerrucci) [inline Gopkg.lock:1144]: where does this new dependency come from ?

File: checks/net_common.go [hunk 3]
```diff
@@ -61,44 +74,60 @@ func createRelationIdentifier(localEndpoint, remoteEndpoint *model.EndpointId, d
...
-	// Reference implementation: https://github.com/weaveworks/scope/blob/master/report/id.go#L40
-	if ipAddress.IsLoopback() {
-		endpoint.Scope = hostname
-	}
 
 	return endpoint
 }
 
-func formatNamespace(clusterName string, n string) string {
+func formatNamespace(clusterName string, hostname string, connection common.ConnectionStats) string {
```
Comment (TizianoPerrucci) [inline checks/net_common.go:99]: I feel we could improve this function which constructs the namespace. 
Right now is adding elements to a list, could we use a more strict system like a struct?
So if easy to add another fragment to the namespace in the future.

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]

File: Gopkg.lock (hunk 2 above)
Comment (craffit) [reply Gopkg.lock:1144]: not sure.. after a go update

**Review by TizianoPerrucci — APPROVED** [review]
cool, thx

**Pull request merged**